objcgen/objcgenerator-processor.cs

```diff
@@ -269,8 +269,9 @@ public override void Process (IEnumerable<Assembly> input)
 					if (processedConstructors.Count > 0)
 						ctors.Add (t, processedConstructors);
 
+					var typeEquals = equals.Where (x => x.Key == t).Select (x => x.Value);
```

Comment:
At some point (hopefully soon) we'll be able to generate the whole BCL (skipping a lot of stuff) which is gonna make an awesome test case to measure performance (and see where any optimization work should happen).